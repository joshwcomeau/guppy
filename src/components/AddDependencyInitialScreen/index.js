// @flow
export { default } from './AddDependencyInitialScreen';
