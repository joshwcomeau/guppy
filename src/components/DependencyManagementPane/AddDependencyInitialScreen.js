// @flow
import React, { Component } from 'react';
import styled from 'styled-components';

import { COLORS } from '../../constants';

import Paragraph from '../Paragraph';
import ExternalLink from '../ExternalLink';
import AlgoliaLogo from './AlgoliaLogo';

type Props = {};

class AddDependencyInitialScreen extends Component<Props> {
  render() {
    return (
      <EmptyState>
        <InstructionsParagraph>
          You can use the input above to search the Node Package Manager (NPM)
          registry for packages that have been published.
        </InstructionsParagraph>
        <InstructionsParagraph>
          Search by package name, description, keyword, or author.
        </InstructionsParagraph>
        <PoweredByWrapper>
          <ExternalLink href="https://www.algolia.com/?utm_source=guppy&amp;utm_medium=website&amp;utm_content=localhost&amp;utm_campaign=poweredby">
            <LinkText>
              <AlgoliaLogo />
            </LinkText>
          </ExternalLink>
        </PoweredByWrapper>
      </EmptyState>
    );
  }
}

export const InstructionsParagraph = Paragraph.extend`
  font-size: 1.4rem;
  color: ${COLORS.lightText};
`;

const EmptyState = styled.div`
  width: 100%;
  height: 100%;
  display: flex;
  flex-direction: column;
  justify-content: space-around;
  padding: 100px 40px;
  text-align: center;
`;

export const PoweredByWrapper = styled.div`
  position: absolute;
  left: 0;
  right: 0;
  bottom: 25px;
`;

const LinkText = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  text-align: center;
  color: ${COLORS.lightText};
`;

export default AddDependencyInitialScreen;
