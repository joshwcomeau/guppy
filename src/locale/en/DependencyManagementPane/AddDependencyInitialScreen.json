{
  "dependencyManagementPane.addDependencyInitialScreen.instruction.searchPackage": "You can use the input above to search the Node Package Manager (NPM) registry for packages that have been published.",
  "dependencyManagementPane.addDependencyInitialScreen.instruction.howToSearch": "Search by package name, description, keyword, or author."
}