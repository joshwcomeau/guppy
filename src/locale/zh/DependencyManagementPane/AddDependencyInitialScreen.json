{
  "dependencyManagementPane.addDependencyInitialScreen.instruction.searchPackage": "您可以使用以上输入框在Node包管理工具NPM中搜索已发布的包。",
  "dependencyManagementPane.addDependencyInitialScreen.instruction.howToSearch": "通过包名称、描述、关键词或作者来搜索。"
}