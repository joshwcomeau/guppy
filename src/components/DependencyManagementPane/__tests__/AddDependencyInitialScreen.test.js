import React from 'react';
import { mount } from 'enzyme';

import AddDependencyInitialScreen, {
  InstructionsParagraph,
  PoweredByWrapper,
} from '../AddDependencyInitialScreen';

jest.mock('../AlgoliaLogo', () => 'svg');

describe('AddDependencyInitialScreen component', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = mount(<AddDependencyInitialScreen />);
  });

  it('should render instruction paragraphs', () => {
    expect(wrapper.find(InstructionsParagraph)).toMatchSnapshot();
  });

  it('should render powered by Algolia link', () => {
    expect(wrapper.find(PoweredByWrapper)).toMatchSnapshot();
  });
});
