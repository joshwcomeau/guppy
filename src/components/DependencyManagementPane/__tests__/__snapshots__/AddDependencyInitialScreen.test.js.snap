// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddDependencyInitialScreen component should render instruction paragraphs 1`] = `
Array [
  .c0 {
  font-size: 1.25rem;
  margin-bottom: 1.5rem;
  font-size: 1.4rem;
  color: #666666;
}

<styled.div>
    <div
      className="c0"
    >
      You can use the input above to search the Node Package Manager (NPM) registry for packages that have been published.
    </div>
  </styled.div>,
  .c0 {
  font-size: 1.25rem;
  margin-bottom: 1.5rem;
  font-size: 1.4rem;
  color: #666666;
}

<styled.div>
    <div
      className="c0"
    >
      Search by package name, description, keyword, or author.
    </div>
  </styled.div>,
]
`;

exports[`AddDependencyInitialScreen component should render powered by Algolia link 1`] = `
.c1 {
  position: relative;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #304FFE;
}

.c1:hover {
  color: #3f6cff;
}

.c0 {
  position: absolute;
  left: 0;
  right: 0;
  bottom: 25px;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  text-align: center;
  color: #666666;
}

<styled.div>
  <div
    className="c0"
  >
    <ExternalLink
      color="#304FFE"
      display="inline-block"
      hoverColor="#3f6cff"
      href="https://www.algolia.com/?utm_source=guppy&utm_medium=website&utm_content=localhost&utm_campaign=poweredby"
    >
      <styled.a
        color="#304FFE"
        hoverColor="#3f6cff"
        href="https://www.algolia.com/?utm_source=guppy&utm_medium=website&utm_content=localhost&utm_campaign=poweredby"
        onClick={[Function]}
        style={
          Object {
            "display": "inline-block",
          }
        }
      >
        <a
          className="c1"
          color="#304FFE"
          href="https://www.algolia.com/?utm_source=guppy&utm_medium=website&utm_content=localhost&utm_campaign=poweredby"
          onClick={[Function]}
          style={
            Object {
              "display": "inline-block",
            }
          }
        >
          <styled.div>
            <div
              className="c2"
            >
              <svg />
            </div>
          </styled.div>
        </a>
      </styled.a>
    </ExternalLink>
  </div>
</styled.div>
`;
